cobbler/tftpgen.py

```diff
@@ -1138,15 +1138,18 @@ def build_kernel_options(
         # This also is architecture specific. E.g: Some ARM consoles need: console=ttyAMAx,BAUDRATE
         # I guess we need a serial_kernel_dev = param, that can be set to "ttyAMA" if needed.
         if system and arch == Archs.X86_64:
-            if system.serial_device or system.serial_baud_rate:
-                if system.serial_device:
-                    serial_device = system.serial_device
-                else:
+            if (
+                system.serial_device > -1
+                or system.serial_baud_rate != enums.BaudRates.DISABLED
+            ):
+                if system.serial_device == -1:
```

Comment:
Wouldn't it be better to set the serial device just to zero per default instead of adding this logic? Or does this have another drawback?

Reply:
This logic is already in there at a different point in the file. This instance of these variables being used was missed in the earlier commit which set the defaults to -1.

`0` is a valid choice for serial_device ('ttyS0'), so I imagine that had something to do with choosing `-1`.

Reply:
Okay thanks for showing me this. I forgot about this indeed.